app/main.py

```diff
@@ -2,5 +2,17 @@
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cache_storage = {}
+
+    def wrapper(*args) -> Callable:
+
+        if args in cache_storage:
+            print("Getting from cache")
+            return cache_storage[args]
+        else:
+            print("Calculating new result")
+            result = func(*args)
+            cache_storage[args] = result
```

Comment:
Your approach is incorrect: you should store arguments and the result of the function with these arguments.